pkg/scheduler/plugins/elasticquota/plugin.go

```diff
@@ -278,6 +280,121 @@ func (g *Plugin) PreFilterExtensions() framework.PreFilterExtensions {
+	unschedulablePodQuotaName, unschedulablePodTreeID := g.getPodAssociateQuotaNameAndTreeID(pod)
+
+	// If unschedulable pod has no quota association, skip
+	// If deleted pod has no quota or different tree, skip
+	if unschedulablePodQuotaName == "" || deletedPodQuotaName == "" || deletedPodTreeID != unschedulablePodTreeID {
+		return framework.QueueSkip
+	}
+
+	return framework.QueueAfterBackoff
+}
+
+// isSchedulableAfterQuotaChanged determines if a pod becomes schedulable after quota is updated.
+// QueueAfterBackoff is default queueingHintFn behavior.
+func (g *Plugin) isSchedulableAfterQuotaChanged(logger klog.Logger, pod *corev1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	originalQuota, modifiedQuota, err := schedutil.As[*apiv1alpha1.ElasticQuota](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "Failed to convert oldObj or newObj to ElasticQuota", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	if originalQuota == nil || modifiedQuota == nil {
+		return framework.QueueAfterBackoff
+	}
+
+	podQuotaName, podTreeID := g.getPodAssociateQuotaNameAndTreeID(pod)
+	if podQuotaName == "" {
+		return framework.QueueSkip
+	}
+
+	// Check if modified quota is in the same tree as the pod
+	modifiedQuotaTreeID := extension.GetQuotaTreeID(modifiedQuota)
+	if modifiedQuotaTreeID != podTreeID {
+		return framework.QueueSkip
+	}
+
+	// Check if modified quota is directly associated with the pod or is an ancestor
+	if modifiedQuota.Name != podQuotaName {
+		// Modified quota is not the pod's quota, check if it's an ancestor
+		if g.pluginArgs.EnableCheckParentQuota {
+			mgr := g.GetGroupQuotaManagerForTree(podTreeID)
+			if mgr == nil {
+				return framework.QueueSkip
+			}
+			podQuotaInfo := mgr.GetQuotaInfoByName(podQuotaName)
+			if podQuotaInfo == nil {
+				return framework.QueueSkip
+			}
+			if !isAncestorQuota(mgr, podQuotaInfo, modifiedQuota.Name) {
+				return framework.QueueSkip
+			}
+		} else {
+			// Parent quota check disabled and not the pod's quota, skip
+			return framework.QueueSkip
+		}
+	}
+
+	// Check if Max or Min increased
+	maxNotIncreased, _ := quotav1.LessThanOrEqual(modifiedQuota.Spec.Max, originalQuota.Spec.Max)
+	minNotIncreased, _ := quotav1.LessThanOrEqual(modifiedQuota.Spec.Min, originalQuota.Spec.Min)
+
+	// If Max or Min increased, allow queueing
+	if !maxNotIncreased || !minNotIncreased {
+		return framework.QueueAfterBackoff
+	}
+
+	// Max and Min didn't increase, check Runtime if enabled
+	if !g.pluginArgs.EnableRuntimeQuota {
+		return framework.QueueSkip
+	}
+
+	oldRuntime, err := extension.GetRuntime(originalQuota)
+	if err != nil {
+		return framework.QueueAfterBackoff
+	}
+
+	newRuntime, err := extension.GetRuntime(modifiedQuota)
+	if err != nil {
+		return framework.QueueAfterBackoff
+	}
+
+	runtimeNotIncreased, _ := quotav1.LessThanOrEqual(newRuntime, oldRuntime)
+	if runtimeNotIncreased {
+		// Runtime not increased, skip
+		return framework.QueueSkip
```

Comment:
hook-plugins may also make the pod unschedulable, we can't compare the resource directly, but can check if quota updated from hook plugins' perspective via GroupQuotaManager#IsQuotaUpdated.